otlp/common.go

```diff
@@ -248,9 +253,36 @@ func getLogsDataset(ri RequestInfo, attrs map[string]interface{}) string {
 	return dataset
 }
 
+// limitedWriter is a writer that will stop writing after reaching its max,
+// but continue to lie to the caller that it was successful.
+// It's a wrapper around strings.Builder for efficiency.
+type limitedWriter struct {
+	max int
+	w   strings.Builder
+}
+
+func newLimitedWriter(n int) *limitedWriter {
+	return &limitedWriter{max: n}
+}
+
+func (l *limitedWriter) Write(b []byte) (int, error) {
+	n := len(b)
+	if n+l.w.Len() > l.max {
```

Comment:
Is the limit in retriever also inclusive?

Reply:
Yes. From hound:
```go
	if stringBytesLen > MaxVarstringLength {
		stringBytes = stringBytes[:MaxVarstringLength]
		stringBytesLen = MaxVarstringLength
	}
```